sanic/signals.py

```diff
@@ -10,13 +10,39 @@
 from sanic_routing.utils import path_to_parts  # type: ignore
 
 from sanic.exceptions import InvalidSignal
+from sanic.log import error_logger, logger
 from sanic.models.handler_types import SignalHandler
 
 
-RESERVED_NAMESPACES = (
-    "server",
-    "http",
-)
+RESERVED_NAMESPACES = {
+    "server": (
+        # "server.main.start",
+        # "server.main.stop",
+        "server.init.before",
+        "server.init.after",
+        "server.shutdown.before",
+        "server.shutdown.after",
+    ),
+    "http": (
+        "http.lifecycle.begin",
+        "http.lifecycle.complete",
+        "http.lifecycle.exception",
+        "http.lifecycle.handle",
+        "http.lifecycle.read_body",
+        "http.lifecycle.read_head",
+        "http.lifecycle.request",
+        "http.lifecycle.response",
+        "http.routing.after",
+        "http.routing.before",
+        "http.lifecycle.send",
+        "http.middleware.after",
+        "http.middleware.before",
+    ),
+}
```

Comment:
future thought: this could be turned into a tree, like:

```python
RESERVED_NAMESPACES = {
    "server": {
        "init": (
            "before",
            "after",
        ),
        "shutdown": (
            "before",
            "after",
        )
    },
    ...
}
```

just for the sake of brevity? I'm just not sure my example is even the best approach ... I mean, that's why the "future thought", in case I'm over complicating this in my head :sweat_smile: